### PR TITLE
processing data in models

### PR DESCRIPTION
**Pull request by fickryiman**
- refactor data models
- rebuild schema database

Create models:
- Create model classes for all entities as shown in the ERD diagram
- Set up associations between models

Use models to insert data:
- Getting familiarize with 'rails c' command
- Open rails console for this project
- Create at least one user
- Create at least 4 posts written by one of the users
- Create at least 6 posts comments for one of the posts
- Use other CRUD methods to find, update and delete entities

Create custom methods:
- Users
   - A method that returns the 3 most recent posts for a given user.
- Posts
   - A method that updates the posts counter for a user.
   - A method which returns the 5 most recent comments for a given post.
- Comments
   - A method that updates the comments counter for a post.
- Likes
   - A method that updates the likes counter for a post.
- Go to rails c and check if all the methods are working.

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -1,4 +1,12 @@
...
+  belongs_to :author, class_name: 'User'
   belongs_to :post
+
+  after_create :update_likes_counter
+
+  private
+
+  def update_likes_counter
+    post.update(likes_counter: post.likes.counter)
+  end
```
Comment (abdulhamiid) [inline app/models/like.rb]: image

Please fix the error in the `Like model` to update the likes counter for a post. 🛠🤗

According to the [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md), when we like to a post, the `likes_counter` for that post should update showing the new number of likes for that post 👍

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @fickryiman :wave, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there! :running_man: :runner: 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Descriptions are very descriptive
- ✔️Set up associations between models.
- ✔️ Write SQL queries with Active Record.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 1 above)
Comment (fickryiman) [reply app/models/like.rb]: OK, there is a typo, it should be post.likes.count instead of post.likes.counter

Comment (fickryiman) [reply app/models/like.rb]: image

**Review by OLIPLICHE — APPROVED** [review]
Hi @fickryiman, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Fixed the typo (✅)) 👏
- Created and sets up associations between models 👏
- Commit messages are meaningful 👏 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**